fs/fat/Kconfig

```diff
@@ -43,7 +43,7 @@ config FAT_LFN
 config FAT_MAXFNAME
 	int "FAT maximum file name size"
 	depends on FAT_LFN
-	default 32
+	default NAME_MAX
```

Comment:
Should be update
```
config NAME_MAX
	int "Maximum size of a file name"
	default 32
	---help---
		The maximum size of a file name.
```
and add range?

Reply:
FAT_MAXFNAME has the spec to define the range, but NAME_MAX is an abstract config, so it's hard to define a range which suitable for all possible file system.

Reply:
Yes. But `FAT_MAXFNAME` has `range 12 255` and if `NAME_MAX` is configured to `300` for example will the Kconfig still limit `FAT_MAXFNAME` to 255? Sorry, my knowledge of Kconfig is limited

Reply:
Yes, the kconfig-frontend saturate FAT_MAXFNAME to 255.